lib/msf/ui/console/command_dispatcher/session.rb

```diff
@@ -0,0 +1,235 @@
+# frozen_string_literal: true
+
+module Msf
+  module Ui
+    module Console
+      module CommandDispatcher
+        module Session
+          include Rex::Ui::Text::DispatcherShell::CommandDispatcher
+
+          @@irb_opts = Rex::Parser::Arguments.new(
+            %w[-h --help] => [false, 'Help menu.' ],
+            '-e' => [true, 'Expression to evaluate.']
+          )
+          def commands
+            {
+              '?' => 'Help menu',
+              'background' => 'Backgrounds the current session',
+              'bg' => 'Alias for background',
+              'exit' => 'Terminate the session',
+              'help' => 'Help menu',
+              'irb' => 'Open an interactive Ruby shell on the current session',
+              'pry' => 'Open the Pry debugger on the current session',
+              'quit' => 'Terminate the session',
+              'resource' => 'Run the commands stored in a file',
+              'uuid' => 'Get the UUID for the current session',
+              'sessions' => 'Quickly switch to another session'
+            }
+          end
+
+          def cmd_background_help
+            print_line('Usage: background')
+            print_line
+            print_line('Stop interacting with this session and return to the parent prompt')
+            print_line
+          end
+
+          def cmd_background(*args)
+            if args.include?('-h') || args.include?('--help')
+              cmd_background_help
+              return
+            end
+            print_status("Backgrounding session #{client.name}...")
+            client.interacting = false
+          end
+
+          alias cmd_bg cmd_background
+          alias cmd_bg_help cmd_background_help
+
+          #
+          # Terminates the session.
+          #
+          def cmd_exit(*args)
+            print_status("Shutting down session: #{client.sid}")
+            client.exit
+          end
+
+          alias cmd_quit cmd_exit
```

Comment:
A session can be `terminated`, `exit'ed`, `shut down` and `quit'ted`. It would be nice to reduce the amount of terms for this action I think.

Reply:
I don't necessarily disagree but I don't think it belongs in this PR, this is just a bit of a refactor and moving things around, the intention is not to alter the way someone would interact with a session here